torch/distributions/transforms.py

```diff
@@ -267,6 +281,12 @@ def inv(self):
             inv._inv = weakref.ref(self)
         return inv
 
+    def with_cache(self, cache_size=1):
+        if self._cache_size == cache_size:
+            return self
+        parts = [part.with_cache(cache_size) for part in self.parts]
```

Comment:
Can we omit this line since this will be done in `__init__`? The calls to `part.with_cache` will return early later, so not too important.

Reply:
Done, thanks for reviewing!